uploadNeuroDB/file_insertion.pl

```diff
@@ -0,0 +1,509 @@
+my @args;
+
+# Describe the usage to be displayed by Getopt::Tabular
+my  $Usage  =   <<USAGE;
+
+This script inserts a file in the files and parameter_file tables.
+
+Usage: perl file_insertion.pl [options]
+
+-help for options
+
+USAGE
+
+# Set the variable descriptions to be used by Getopt::Tabular
+my $profile_desc       = "name of config file in ./dicom-archive/.loris_mri.";
+my $file_path_desc     = "file to register into the database (full path from "
+                         . "the root directory is required)";
+my $upload_id_desc     = "ID of the uploaded imaging archive containing the "
+                         . "file given as argument with -file_path option";
+my $pname_desc         = "patient name, if cannot be found in the file name "
+                         . "(in the form of PSCID_CandID_VisitLabel)";
+my $output_type_desc   = "file's output type (e.g. native, qc, processed...)";
+my $scan_type_desc     = "file's scan type (from the mri_scan_type table)";
+my $date_acquired_desc = "acquisition date for the file (YYYY-MM-DD)";
+my $scanner_id_desc    = "ID of the scanner stored in the mri_scanner table";
+my $coordin_space_desc = "Coordinate space of the file to register (e.g. "
+                         . "native, linear, nonlinear, nativeT1)";
+my $reckless_desc      = "upload data to database even if study protocol is "
+                         . "not defined or violated.";
+my $metadata_file_desc = "file that can be read to look for metadata "
+                         . "information to attach to the file to be inserted";
+
+
+# Initialize the arguments table
+my @args_table = (
+
+    ["Mandatory options", "section"],
+
+        ["-profile",       "string",  1, \$profile,       $profile_desc      ],
+        ["-file_path",     "string",  1, \$file_path,     $file_path_desc    ],
+        ["-upload_id",     "string",  1, \$upload_id,     $upload_id_desc    ],
+        ["-output_type",   "string",  1, \$output_type,   $output_type_desc  ],
+        ["-scan_type",     "string",  1, \$scan_type,     $scan_type_desc    ],
+        ["-date_acquired", "string",  1, \$date_acquired, $date_acquired_desc],
+        ["-scanner_id",    "string",  1, \$scanner_id,    $scanner_id_desc   ],
+        ["-coordin_space", "string",  1, \$coordin_space, $coordin_space_desc],
+
+    ["Advanced options", "section"],
+
+        ["-reckless",    "boolean", 1, \$reckless,    $reckless_desc ],
+        ["-verbose",     "boolean", 1, \$verbose,     "Be verbose"   ],
+
+    ["Optional options", "section"],
+        ["-patient_name",  "string", 1, \$patient_name,  $pname_desc        ],
+        ["-metadata_file", "string", 1, \$metadata_file, $metadata_file_desc]
+
+);
+
+Getopt::Tabular::SetHelp ($Usage, '');
+##TODO 1: replace exit 1 by $NeuroDB::ExitCodes::GETOPT_FAILURE
+GetOptions(\@args_table, \@ARGV, \@args) || exit 1;
+
+# Input option error checking
+if  (!$profile) {
+    print "$Usage\n\tERROR: You must specify a profile.\n\n";
```

Comment:
Print to STDERR maybe, since that's a major error. There are also more occurences of cases like this below.

Reply:
ooops. I forgot :S. Done now